routers/api/v1/repo/issue_label.go

```diff
@@ -310,9 +342,43 @@ func prepareForReplaceOrAdd(ctx *context.APIContext, form api.IssueLabelsOption)
 		return
 	}
 
-	labels, err = models.GetLabelsByIDs(form.Labels)
-	if err != nil {
-		ctx.Error(http.StatusInternalServerError, "GetLabelsByIDs", err)
+	switch form.Mode {
+	case "name_only":
+		labels, err = models.GetLabelsInRepoByNames(ctx.Repo.Repository.ID, form.Labels)
+		if err != nil {
+			ctx.Error(http.StatusInternalServerError, "GetLabelByIDsOrNames", err)
+			return
+		}
+	case "mixed":
```

Comment:
What use case does `mixed` cover? I can't think of any. I'd just go with modes `"id"` & `"name"`